src/components/PlanRoom/RoomTables/Item.tsx

```diff
@@ -0,0 +1,184 @@
+import { useCallback, useState, SetStateAction } from 'react';
+import { Box, Textarea } from '@chakra-ui/react';
+import update from 'immutability-helper';
+import { useDrop, XYCoord, useDrag } from 'react-dnd';
+import ItemTypes from './ItemTypes';
+import { TypeItem, TableType } from './Types';
+import SaveBoard from '../SaveRoomTables/SaveBoard';
+
+let changeValueTable: Function;
+let changeValueTableBorderRadius: Function;
+let newTable: Function;
+let save: Function;
+const backgroundColor: string = '#211F1F';
+
+const Items = () => {
+  const [tables, setTables] = useState<{
+    [key: string]: { top: number; left: number; text: string; width: number; height: number; borderRadius: number };
+  }>({
+    0: { top: 0, left: 0, text: '0', width: 100, height: 50, borderRadius: 0 },
+  });
+
+  const moveTable = useCallback(
```

Comment:
You can merge those three functions one below the other to one by passing optional parameters or in some other way.
You do the same thing three times except for passing different parameters.

Reply:
Right, but I'll leave it at that for now and move on